### ISSUE TITLE
Define a `template-include` directive

### ISSUE THREAD

**Issue by gdemonet** [scality/sphinx-tools#31]
This allows to define a Jinja template and render it into RST before including it in a procedure.
Very useful for building documentation from sub-projects exposing their "include-able" fragments.

**Comment by wabernatScality** [comment]
I want to learn more!

**Comment by gdemonet** [comment]
I will make sure to demo all this in person with your team @wabernatScality :wink: 

Until then, here's a glimpse of the concept:

- Assume there is some re-usable piece of documentation, but one that isn't really static (each time you include this piece, you'd like to change a few details). For instance, let's assume we have a piece describing how to put a server into maintenance mode. Obviously, the server's name is something that needs to change depending on the context of inclusion.

- The first step will be to build a template of this piece. This is basically standard reST syntax, with some sprinkles of Jinja (which you can find documented [here](https://jinja.palletsprojects.com/en/2.11.x/templates/)). Here is how it could look like:

```jinja
Put the server "{{ name }}" server in maintenance mode:

.. code:: console

   {{ prompt | default('root #'}} /srv/scality/awesome-tool enable-maintenance --server={{ name }}
```

- Assuming this template is available in some configured directory, and called `server-maintenance.rst.j2` (we usually append `.j2` to filenames to indicate Jinja **2** templating), it can now be included in any procedure which needs it. For instance, in a "single-node cluster maintenance" topic, where the server is always called "bootstrap", this would become:

```rst
Single Node Cluster Maintenance
===============================

.. template-include:: server-maintenance.rst.j2

   {"name": "bootstrap"}
```

Behind the scenes, this inclusion will render the template into a final reST document, which would then be fed to Sphinx for final (HTML/LaTeX/whatever you need) rendering. This reST document would be:

```rst
Single Node Cluster Maintenance
===============================

Put the server "bootstrap" server in maintenance mode:

.. code:: console

   root # /srv/scality/awesome-tool enable-maintenance --server=bootstrap
```

If you had instead this source:

```rst
Three Nodes Cluster Maintenance
===============================

.. template-include:: server-maintenance.rst.j2

   {"name": "bootstrap", "prompt": "[root@bastion]#"}


.. template-include:: server-maintenance.rst.j2

   {"name": "master-1", "prompt": "[root@bastion]#"}


.. template-include:: server-maintenance.rst.j2

   {"name": "master-2", "prompt": "[root@bastion]#"}
```

This would render to (note the prompt change):

```rst
Three Nodes Cluster Maintenance
===============================

Put the server "bootstrap" server in maintenance mode:

.. code:: console

   [root@bastion]# /srv/scality/awesome-tool enable-maintenance --server=bootstrap

Put the server "master-1" server in maintenance mode:

.. code:: console

   [root@bastion]# /srv/scality/awesome-tool enable-maintenance --server=master-1

Put the server "master-2" server in maintenance mode:

.. code:: console

   [root@bastion]# /srv/scality/awesome-tool enable-maintenance --server=master-2
```

(of course, this would need to be refined, for instance including the piece as a list item)

An easy improvement that can be added to this feature is to allow repetition of inclusions. I built a prototype of this, which works as follows (exact same result as the example above):

```rst
Three Nodes Cluster Maintenance
===============================

.. template-include:: server-maintenance.rst.j2

   [
       {"name": name, "prompt": "[root@bastion]#"}
       for name in ["bootstrap", "master-1", "master-2"]
   ]
```

Basically, you have all the power of Python expressions to generate rendering contexts, which will be used for rendering the included pieces you need in the final procedure.

Hope you like it!
